Assets/Scripts/Inventory/Inventory.cs

```diff
@@ -2,65 +2,148 @@
 using System.Collections.Generic;
 using UnityEngine;
 
-// This class represents a slot within the player's inventory that holds an item
-[System.Serializable]
-public class ItemSlot
+namespace FunMath
 {
-    public ItemSlot(Item item, int Count)
+    // This class represents a slot within the player's inventory that holds an item
+    [System.Serializable]
+    public class ItemSlot
     {
-        this.item = item;
-        this.ItemCount = Count;
+        public ItemSlot(Item item, int Count)
+        {
+            this.item = item;
+            this.ItemCount = Count;
+        }
+        [SerializeField]
+        public Item item;
+        public int ItemCount = 0;
+        public bool Empty()
+        {
+            return ItemCount == 0;
+        }
     }
-    public Item item;
-    public int ItemCount = 0;
-}
-
-public class Inventory : MonoBehaviour
-{
-    public int MaxNumberOfItem = 100;
-    public bool InfiniteItem = false;
 
-    [SerializeField]
-    private List<ItemSlot> Items = new List<ItemSlot>(); // Holds the player's collected items
-    public List<ItemSlot> GetInventoryItemSlots()
+    public class Inventory : MonoBehaviour
     {
-        return Items;
-    }
+        public int MaxNumberOfItem = 100;
+        public bool InfiniteItem = false;
+        public int MaxNumberOfTypes = (int)Item.ItemType.OperationMax; // By default goes to max number of operations
 
-    public void ModifyItemSlot(Item item, int count)
-    {
-        ItemSlot slot = FindItem(item.thisItemType);
+        private void Start()
+        {
+            Items = new List<ItemSlot>(MaxNumberOfTypes);
+            // Regardless of items, we will always have this amount of inventory slots
+            for(int i = 0; i < MaxNumberOfTypes; ++i)
+            {
+                Items.Add(new ItemSlot(null, 0));

```

Comment:
How can we create this so it always has the default values?  Like we always have `+`/`-`

Reply:
I was imagining the game manager would give those default values somewhere. It'll just add `+`/'-' in.